article/app/controllers/LiveBlogController.scala

```diff
@@ -259,19 +268,30 @@ class LiveBlogController(
     val newBlocks = getNewBlocks(page, lastUpdateBlockId, filterKeyEvents)
     val newCapiBlocks = getNewBlocks(requestedBodyBlocks, lastUpdateBlockId, filterKeyEvents)
 
-    val timelineHtml = views.html.liveblog.keyEvents(
-      "",
-      model.KeyEventData(newBlocks, Edition(request).timezone),
-      filterKeyEvents,
-    )
-
     for {
       blocksHtml <-
         if (remoteRender) {
-          getDCRBlocksHTML(page, newCapiBlocks)
+          // Only query DCR if we know there are updates
```

Comment:
If we have new blocks will we make 2 extra calls to DCR?

Reply:
I think right now we are always making 1 call to DCR for every query

Now we will make 0 when there are no new blocks, and 2 when there are new blocks :)

Reply:
I wonder if we can optimise this process:
0 if no new blocks
1 if new blocks, eg

`(blocksHtml, timelineHtml) <- getHtml`

Where getHtml passes the new blocks to DCR, DCR makes the decision about rendering key events html and blocks html, returning a response combining both.

In the current flow it feels slightly strange that we're making 2 network requests based on the same decision (`if (newBlocks.nonEmpty)`). However, I might be missing some extra context about the decisions made

Reply:
I definitely agree with would be better;

Up until now (at least to my understanding), DCR has only ever returned HTML directly from its API, so to facilitate this improvement I think we'd need to update DCR to return a JSON data model for an endpoint, making it look something like this

```json
// POST (DCR)/Live-update
{
     "blocks": "<p>some HTML</p>
     "keyEvents": "<p> some HTML</p>"
}
```

I think technically, this wouldn't be a difficult change, returning JSON from an express server isn't too hard, I'd just want to make sure that returning JSON data instead of raw HTML wouldn't some kind of line that DCR wasn't supposed to cross.

Reply:
Curious how @oliverlloyd feels about the above ^^